src/tools/rbd_mirror/image_replayer/snapshot/Replayer.cc

```diff
@@ -196,6 +196,15 @@ void Replayer<I>::shut_down(Context* on_finish) {
 
     // TODO interrupt snapshot copy and image copy state machines even if remote
     // cluster is unreachable
```

Comment:
So, isn't the problem actually due to this "TODO" is still not implemented? And the right fix would be to implement it?

Reply:
Yes the driver of this patch is that the "TODO" isn't implemented. The approach in my patch implements it via a trigger + image_replayer stop. I was attempting to address force promote with mostly code and functions that already existed today. While I reconsider the approach, can you provide any additional thoughts?